### DISCUSSION TITLE
Bad ELM response: WRONG RESPONSE : NR: Request Out Of Range(7F2231)

### DISCUSSION THREAD

**Discussion by x5nder** [cedricp/ddt4all#1020] [Q&A]
Before I'm going to try and convince a Renault dealership to unlock the CGW on my Clio V with CLIP, I wanted to do a few test runs so I know where to find all parameters that I want to change in ddt4all. So when I connect, select the correct car (CLIO V), select one of the ecus (A-IVI v1.10) and then the correct screen (Navigation > System Information $DAB1), I get actual read-out values in green, but the bottom log window of ddt4all is giving me tons of "Bad ELM response: WRONG RESPONSE : NR: Request Out Of Range(7F2231)" errors. Is this normal, or am I doing something wrong? I am assuming the read-out values in green mean that the ELM327 scanner is working properly and that the errors either mean specific parts of the ecu can't be read, or it's because the CGW is not unlocked yet... am I mistaken?

**Comment by AndyIreland** [comment]
could be the gateway ..... but why are you, or it sounds like you are , selecting the ECU's manually , Why dont you try searching for the ECU's in your car with the magnifying glass in DDT4ALL?

**Comment by x5nder** [comment]
Hmm. You might be onto something here. If I do a scan, I'm getting (in green) the following ecus:

[VSP] VSP_v1a3
[MIU] Central Panel CMFB/CDp2_7/9.3_2
[USM[ USM_C1A_SW6
[HFM] C1A_HFM_DDT_7_1
and one or two more (can't remember)...
...but none of the [Navigation] ecus. 

The last Navigation ecu 'version' for my Clio is A-IVI v1.12... does this mean I should find a newer version somewhere, or should I draw a different conclusion? The ecus in green can be read fine according to the log ('ECU uses SDS 10xx' and 'Switching to session mode 10xx') but when I try to access the Navigation ecu, that's when I'm getting the error mentioned in the title of this topic.

**Comment by x5nder** [comment]
Hmmm. I found another ecu update where the latest Navigation ecu version is v2.8, but even here none of the navigation ecus are recognized. I know that my EasyLink head unit was very recently updated to the latest version; could it be that there is no ecu.zip file yet that has the correct Navigation ecu version for the current EasyLink software version installed in my car?

**Comment by AndyIreland** [comment]
yeah 2.8 is the latest I can see on mine too:
![image](https://github.com/cedricp/ddt4all/assets/49172217/84dac9d5-f656-4fc3-a3e9-cfc769cb3127)

I wonder if for theClio5 you still have to make up a special splitter cable or solder the ELM327 to read CAN2 / CAN HI like you did with the earlier radionav's or whether thats eliminated on the Clio 5 now? 

When I have seen people 'apparently' bypassing the security gateway that have took apart the dash and wired the ELM327 directly to the back of the BCM or nav wiring to change parameters on the radionav

https://myrenaultclioblog.wordpress.com/2023/05/22/add-wireless-android-auto-wireless-car-play-to-renault-clio-mk5-bypassing-the-security-gateway/

**Comment by x5nder** [comment]
I found my old screenshots, and in those I can see that I actually had access to A-IVI v1.10 without errors until 2 weeks ago. I am going to assume that the software update performed by the Renault dealer last week (to v2.13 / 04-2023) updated the Navigation ecu to a version that's not (yet) supported by the existing ecu databases.

Now the main question is: is there a way to obtain updated ecu files? Where do they come from? Are they extracted from the official CLIP software? I am assuming all current Clio VI cars (ie, the 2024 model) will run on the same updated ecu version, so at some point there has to be support for them.

Maybe this is even a question for @cedricp as he's most intimately knowledgeable on this topic.

**Comment by AndyIreland** [comment]
OK, so I cannot find this article I read (typical!) and i have been in my history and bookmarks. There was a post somewhere and I am pretty sure it was for the easylink , that said something like: 
"If you are wanting to enable Wireless Android Auto & Apple Car Play be sure to do it before update so and so" because from what I made out of that is Renault has updated the software/firmware of the easylink to close the loop hole of people being able to change the settings AFTER this update!! - so it could be that. 

Also Cedric P program (DDT4ALL) does not even touch the ECU Database, in fact originally you had to download DDT4ALL originally and source your ecu.zip file from other sources (namely latest DDT2000 Database) so i dont think that will help. - I cannot see you even getting a more up to date database as well , possibly you might but I think the problem is not the actual database , but the fact they done the software update on your easylink unit to v2.13 and that has closed up that loophole that was there on the version before it.

**Comment by x5nder** [comment]
Ahhhh, damnit... I never read that piece of information, otherwise I would have made sure to take care of this before the update. Ah well... nothing that can be done about that (pretty sure it's not going to be easy or even possible at all to roll back to a previous version). Thanks for the info, though...!!!!

**Comment by AndyIreland** [comment]
Thats OK - sorry if I give you wrong information though. You know there is a possibility I got / supply the wrong info, so if I have I am very sorry about that. 

what is  it you are looking to do / enable/ disable on your easylink anyway particularly or just change some different parameters?

With the latest software Update I think it gives you wireless android auto feature now as standard (not wireless apple car play though I dont think) you just have to yurn it on in settings menu of the easylink

**Comment by x5nder** [comment]
No worries-- it definitely looks like Renault disabled access to the Navigation ecu in their 2023 updates. I should have been smart enough to get the CGW unlocked first to make the changes I wanted, before letting Renault update my system.

I was mainly looking into:
* enabling wireless Carplay (even though I'm currently using a wireless dongle that works quite well);
* changing L/100km to km/L;
* enabling MEX (multi-sense) so I can enable Eco mode as default;
* change the startup logo.

But all of these are done on the Navigation parameter page, so... tough luck. 

I might try to install CLIP itself (found a cracked version from September 2023) to see if that will offer me any opportunities, but I seriously doubt it.

**Comment by KarelSvo** [comment]
[here](https://www.clio5-forum.de/forum/thread/976-ddt4all-gateway-router-mod/) and[ here](https://www.clio5-forum.de/forum/thread/955-ddt4all-clio-v-mod/)

https://github.com/KarelSvo?tab=repositories

**Comment by x5nder** [comment]
Thanks @KarelSvo but it doesn't answer my main question-- did Renault block access to the navigation ecu in its latest Easy Link versions, and if so, can it be reverted or circumvented?

**Comment by KarelSvo** [comment]
Since you didn't provide any log files, nothing can be said about your problem.
I recommend using PyRen, it has a backup function included.

**Comment by x5nder** [comment]
@KarelSvo apologies, I was on a short holiday. Attached are the ddt4all log files, and pasted below is the Python output from ddt4all. As you can see, it recognizes a Navigation ecu, but can't find an existing ecu version in the database.

```
Found ECU [BCM/UCH] (not perfect match) :C1A_BCM_DDT_7_0_2 DIAGVERSION [90] SUPPLIER [CONTINENTAL AUTOMOTIVE] SOFT [142902 ] VERSION [0710 instead 0800 ]
Identified ECU [UPC-EMM]@27 : USM_C1A_SW6_20190527T111154.json DIAGVERSION [64]SUPPLIER [414] SOFT [BA65] VERSION [6560] {33663}
Found ECU [HVAC] (no relevant ECU file found) :DIAGVERSION [09] SUPPLIER [B517] SOFT [SW16.C.01-B073-1] VERSION [F00L423]
Identified ECU [FRRAD]@2B : RADAR_C1A_v8.1_20190823T144542.json DIAGVERSION [08]SUPPLIER [1L9] SOFT [09.00.02] VERSION [2019CW08] {12623}
Found ECU [Airbag] (no relevant ECU file found) :DIAGVERSION [04] SUPPLIER [D321 ] SOFT [SW_S30M_RNI_C1A_] VERSION [0642190820 ]
Identified ECU [MIU]@59 : Central_Panel_CMFB_CDp2_7_9.3_20201016_20201019T151954.json DIAGVERSION [08]SUPPLIER [LGE_VIETNA] SOFT [SW07.10] VERSION [SW07.10] {20179}
Found ECU [Navigation] (no relevant ECU file found) :DIAGVERSION [B1] SUPPLIER [BOSCH] SOFT [AI_PRJ_RN_AIVI_2] VERSION [6215_230630]
Found ECU [Cluster/TDB] (not perfect match) :Cluster C1A_RUN1_V2.29 DIAGVERSION [12] SUPPLIER [07-Visteon-Namestovo ] SOFT [V05.00] VERSION [1951 instead 1951]
Identified ECU [VSP]@81 : VSP_v1a4_20200211T125058.json DIAGVERSION [4]SUPPLIER [U2F] SOFT [0005] VERSION [0004] {3626}
Found ECU [Power steering] (no relevant ECU file found) :DIAGVERSION [18] SUPPLIER [4AG4] SOFT [D100_106.22.01.0] VERSION [106.22.01.09 ]
Found ECU [EPS] (no relevant ECU file found) :DIAGVERSION [18] SUPPLIER [4AG4] SOFT [D100_106.22.01.0] VERSION [106.22.01.09 ]
Identified ECU [AVM]@61 : Entry_AVM_Renault_08_10_2021_20211011T185442.json DIAGVERSION [4]SUPPLIER [!4`] SOFT [0030] VERSION [0014] {401}
Found ECU [FCAM] (not perfect match) :FrontCamera_C1A_V3.18 DIAGVERSION [08] SUPPLIER [AFK] SOFT [SW0ST13] VERSION [1334388612000000 instead 13363B5412060100]
Identified ECU [HFM]@C0 : C1A_HFM_DDT_8_20190802_20190806T144254.json DIAGVERSION [05]SUPPLIER [CONTINENTAL] SOFT [142A02 ] VERSION [0620 ] {6275}
```

[ecu_ddt.txt](https://github.com/cedricp/ddt4all/files/14232152/ecu_ddt.txt)
[elm_ddt.txt](https://github.com/cedricp/ddt4all/files/14232154/elm_ddt.txt)
[screens.txt](https://github.com/cedricp/ddt4all/files/14232155/screens.txt)

**Comment by KarelSvo** [comment]
Your log data only shows scanning for control devices. Not your problem from the topic title.

**Comment by x5nder** [comment]
Well, it shows that it can't find a matching json file for the Navigation ECU, even though I have the same A-IVI v0.9 - A-IVI v2.8 files as AndyIreland mentioned above. If you select _any_ non-identified ECU in the top right box (either for your car, or for a random other car), then try to open the details in the middle box, you will get the Request Out Of Range error that I mentioned in the topic.

**Reply by pyren4all** [reply]
Hello @x5nder 
Were you able to resolve this issue ? I have the same issue on my XJAph2 and cannot find a solution. If you have a fix, please let me know, that would be greatly appreciated. Thank you !

**Comment by KarelSvo** [comment]
A ddt2000 file from 2022 cannot contain Autoident data from 2023.
But that's not a problem, you just use the most current file available.
If you do not provide log data and screenshots of 'Bad ELM response:', this message cannot be identified.
No menu was opened in your log files shown, only control devices were searched.

**Comment by estyasser** [comment]
I'm facing the same issue. I get this message when I try to load the screen (RadNav configuration 2.2) - as I was attempting to do [this tutorial to enable RS Monitor](https://www.drive2.ru/l/590280515026719804/?page=0#comments)
I have the [latest ddt4all version](https://github.com/cedricp/ddt4all/releases/tag/v3.0.0) installed and the ecu is downloaded from [this reply](https://github.com/cedricp/ddt4all/discussions/845#discussion-5264316)
I've tried different versions of ddt4all, and they all produce the same issue, even the mobile app (ECUTweaker)

My car is 2017 Renault Talisman TCe 160 with R-Link2 v3.3.16.986
My audio unit was replaced from a totaled Renault Megane IV Sedan - which means my car VIN is difference from the Audio Unit VIN. Could that be the problem? If yes, is there a way to re-write the VIN?

`ECU uses SDS 1003
Bad ELM response : WRONG RESPONSE : NR: Request Out Of Range(7F2231)
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 
Sending ELM request :3E00
ELM response : 7E 00 `
